pkg/ide/jupyter.go

```diff
@@ -23,82 +20,204 @@ import (
 
 const startJupyterCommand = "notebook --no-browser --port=8888 --ip=0.0.0.0 --NotebookApp.token='' --NotebookApp.password=''"
 
-func OpenJupyterIDE(activeProfile config.Profile, workspaceId string, projectName string, projectProviderMetadata string) error {
-	// Download and start IDE
+// OpenJupyterIDE manages the installation and startup of a Jupyter IDE on a remote workspace.
+func OpenJupyterIDE(activeProfile config.Profile, workspaceId, projectName, projectProviderMetadata string) error {
+	// Ensure SSH config entry is added
 	err := config.EnsureSshConfigEntryAdded(activeProfile.Id, workspaceId, projectName)
 	if err != nil {
 		return err
 	}
 
-	views.RenderInfoMessageBold("Installing Jupyter Notebook...")
 	projectHostname := config.GetProjectHostname(activeProfile.Id, workspaceId, projectName)
 
+	// Check and install Python if necessary
+	if err := ensurePythonInstalled(projectHostname); err != nil {
+		return err
+	}
+
+	// Check and install pip if necessary
+	if err := ensurePipInstalled(projectHostname); err != nil {
+		return err
+	}
+
 	// Install Jupyter Notebook
-	installJupyterCommand := exec.Command("ssh", projectHostname, "python3 -m pip install --user notebook && export PATH=$HOME/.local/bin:$PATH && echo $PATH")
-	installJupyterCommand.Stdout = io.Writer(&util.DebugLogWriter{})
-	installJupyterCommand.Stderr = io.Writer(&util.DebugLogWriter{})
+	if err := installJupyter(projectHostname); err != nil {
+		return err
+	}
 
-	err = installJupyterCommand.Run()
-	if err != nil {
-		return fmt.Errorf("failed to install Jupyter Notebook: %w", err)
+	// Start Jupyter Notebook server
+	if err := startJupyterServer(projectHostname, activeProfile, workspaceId, projectName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ensurePythonInstalled checks if Python is installed and installs it if the user agrees.
+func ensurePythonInstalled(hostname string) error {
+	views.RenderInfoMessageBold("Checking Python installation...")
+
+	// Check if Python is installed
+	if err := runRemoteCommand(hostname, "python3 --version"); err != nil {
+		log.Error("Python3 is not installed on the remote workspace.")
+
+		// Prompt user to install Python
+		fmt.Print("Would you like to install Python3? (yes/no) [yes]: ")
+		var userInput string
+		fmt.Scanln(&userInput)
```

Comment:
Please use a TUI component for this (e.g. https://github.com/daytonaio/daytona/blob/main/pkg/views/purge/confirm.go)

Also, the user should be able to pass a `--yes/-y' to the invoking command to automatically select yes here so they don't have to interact with the TUI

Reply:
For the flag part, you can check an example here https://github.com/daytonaio/daytona/blob/main/pkg/cmd/server/server.go#L32